NickvisionTubeConverter.Shared/Controllers/AddDownloadDialogController.cs

```diff
@@ -293,22 +293,21 @@ public void PopulateDownloads(MediaFileType mediaFileType, Quality quality, int?
     /// <param name="audioLanguage">The audio language code</param>
     /// <param name="subtitles">Whether or not to download the subtitles</param>
     /// <param name="saveFolder">The save folder of the downloads</param>
-    /// <param name="limitSpeed">Whether or not to use speed limit</param>
-    /// <param name="splitChapters">Whether or not to split based on chapters</param>
-    /// <param name="cropThumbnail">Whether or not to crop the thumbnail</param>
-    /// <param name="timeframe">A Timeframe to restrict the timespan of the media download</param>
-    /// <param name="credentialIndex">The index of the credential to use</param>
-    public async Task PopulateDownloadsAsync(MediaFileType mediaFileType, Quality quality, int? resolution, string? audioLanguage, bool subtitles, string saveFolder, bool limitSpeed, bool splitChapters, bool cropThumbnail, Timeframe? timeframe, int credentialIndex)
+    /// <param name="options">AdvancedDownloadOptions</param>
+    public async Task PopulateDownloadsAsync(MediaFileType mediaFileType, Quality quality, int? resolution, string? audioLanguage, bool subtitles, string saveFolder, AdvancedDownloadOptions options, int credentialIndex)
     {
         if (_keyring != null)
         {
             var credentials = await _keyring.GetAllCredentialsAsync();
             credentials.Sort((a, b) => a.Name.CompareTo(b.Name));
-            PopulateDownloads(mediaFileType, quality, resolution, audioLanguage, subtitles, saveFolder, limitSpeed, splitChapters, cropThumbnail, timeframe, credentials[credentialIndex].Username, credentials[credentialIndex].Password);
+            options.Username = credentials[credentialIndex].Username;
+            options.Password = credentials[credentialIndex].Password;
         }
         else
         {
-            PopulateDownloads(mediaFileType, quality, resolution, audioLanguage, subtitles, saveFolder, limitSpeed, splitChapters, cropThumbnail, timeframe, "", "");
+            options.Username = null;
+            options.Password = null;
         }
```

Comment:
Yes but because the `options` object is passed in from somewhere else, we don't know what they could have set it to. And because this is a failed case where the keyring doesn't exist, it's best to set it null.